! A. # Ressources

- blog
https://flaven.fr/

- youtube videos
https://www.youtube.com/channel/UCnUBoVx9Yai3wirPBvNpNQw


- 1 -  all code
https://github.com/bflaven

- 2 - learning_python_for_all
--- https://github.com/bflaven/learning_python_for_all
A set of resources from Books, Exercises, Notebooks to discover and learn Python. It exists both in French and in English.


- 3 - learning_python_for_all
--- https://github.com/bflaven/new_powers_new_responsibilities_the_journalism_ai_report

Un certain nombre de définition utiles à l’IA sont disponibles dans le doc GitHub - bflaven/new_powers_new_responsibilities_the_journalism_ai_report: Une synthèse d'un excellent document disponible en ligne, relativement récent (2019) extrait du "The JournalismAI Report", nommé "New powers, new responsibilities. A global survey of journalism and artificial intelligence", de Charlie Beckett, Professeur à la London School of Economics and Political Science. 



! B. #  Using LLM

Installation LM studio. Only available for LM Studio for Mac (M1/M2/M3)

- 1.  Tool LM Studio
https://lmstudio.ai/

- 2. ollama
https://ollama.ai/

# COMMANDS FOR OLLAMA

# To run and chat with Llama 2
ollama run llama2
ollama run llama2-uncensored
ollama run orca-mini


# remove a model
ollama rm llama2
ollama rm orca-mini
ollama rm mistral
ollama rm falcon:7b
ollama rm mistral:text
ollama rm llama2:latest
ollama rm orca-mini:latest

# list the model
ollama list

# when you are in the model you can use
>>> /?
>>> /list
>>> /set verbose

# to get out from the model
/exit




--- for LLMs huggingface.co
https://huggingface.co/



! C. #  Addons for sublime and visual code
--- Best plugins for Sublime Text
Package control (https://packagecontrol.io/)
SideBar Enhancements
Sublime CodeIntel
Emmet
Bracket Highlighter
Git
Git Gutter
Sublime Linter
DocBlockr
GoSublime
Browser Refresh
Auto Semi-Colon
Advanced New File
LaTeX Tools
Quick File Creator
SublimeREPL
Color Highlighter
JSLint
SFTP
Pretty JSON


--- Best plugins for Visual Code
- productivity_1
Prettier
Remote – SSH
Live Share
Better Comments
CodeSnap
Code Runner
Path Intellisense
vscode-icons
Night Owl

- productivity_2
Live Server
Auto Rename Tag
SVG Preview
HTML CSS Support
IntelliSense for CSS class names in HTML
CSS Peek
GitLens

- js
ESLint
Extraits de code JavaScript (ES6)
Quokka.js
npm Intellisense
Import Cost

- Python
Python
Pylance
Jupyter
Django
Flask Snippets

- development experience
GitHub Copilot
Tabnine AI Autocomplete Tabnine AI
Markdown All in One
Regex Previewer


! D. TOOLS & ENVIRONMENT

- Tools to create and manage of virtual environments:

--- ANACONDA
https://www.anaconda.com/

--- PYTHON-POETRY
https://python-poetry.org/

--- VENV
https://docs.python.org/3/library/venv.html

--- MANAGE ENVIRONMENT ANACONDA
[env]
# Conda Environment
conda create --name langchain_ai python=3.9.13
conda info --envs
source activate langchain_ai
conda deactivate

# if needed to remove
conda env remove -n [NAME_OF_THE_CONDA_ENVIRONMENT]
conda env remove -n langchain_ai

# update conda 
conda update -n base -c defaults conda

# to export requirements
pip freeze > requirements.txt

# to install
pip install -r requirements.txt

conda install -c conda-forge langchain


# [path]
cd /Users/brunoflaven/Documents/03_git/ia_usages/ia_using_other_llm/


# launch the file
python samwit_basic_chain.py